### PR TITLE
Move discovery.initial_state_timeout from DiscoverySettings to Node

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this improves CrateDB

We decided to keep the `DiscoverySettings` for now as deprecated, this removes it only from the usage. 

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/node/Node.java [hunk 1]
```diff
@@ -210,6 +209,9 @@
         )
     );
 
+    public static final Setting<TimeValue> INITIAL_STATE_TIMEOUT_SETTING =
```
Comment (mfussenegger) [inline server/src/main/java/org/elasticsearch/node/Node.java:212]: Shouldn't this be removed from `DiscoverySettings` ?

**Review by mfussenegger — APPROVED** [review]
lgtm, although I wonder if we should go with a different commit message. Given that we still keep the settings around the `Remove zen1` is misleading.

**Comment by mkleen** [comment]
> lgtm, although I wonder if we should go with a different commit message. Given that we still keep the settings around the `Remove zen1` is misleading.

Maybe something more like "Move ``discovery.initial_state_timeout`` from DiscoverySettings to Node" without a backport ?

**Comment by mfussenegger** [comment]
> > lgtm, although I wonder if we should go with a different commit message. Given that we still keep the settings around the `Remove zen1` is misleading.
> 
> Maybe something more like "Move `discovery.initial_state_timeout` from DiscoverySettings to Node" without a backport ?

yep :+1:

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please

**Pull request merged**